File: src/test/java/br/com/jiratorio/factory/domain/request/LeadTimeConfigRequestFactory.java
package br.com.jiratorio.factory.domain.request;

import br.com.leonardoferreira.jbacon.JBacon;
import br.com.jiratorio.domain.request.LeadTimeConfigRequest;
import com.github.javafaker.Faker;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class LeadTimeConfigRequestFactory extends JBacon<LeadTimeConfigRequest> {

    @Autowired
    private Faker faker;

    @Override
    protected LeadTimeConfigRequest getDefault() {
        LeadTimeConfigRequest leadTimeConfigRequest = new LeadTimeConfigRequest();

        leadTimeConfigRequest.setName(faker.lorem().word());
        leadTimeConfigRequest.setStartColumn(faker.lorem().word());
        leadTimeConfigRequest.setEndColumn(faker.lorem().word());

        return leadTimeConfigRequest;
    }

    @Override
    protected LeadTimeConfigRequest getEmpty() {
        return new LeadTimeConfigRequest();
    }

    @Override
    protected void persist(final LeadTimeConfigRequest leadTimeConfigRequest) {
        throw new UnsupportedOperationException();
    }
}

File: src/test/kotlin/br/com/jiratorio/factory/domain/request/LeadTimeConfigRequestFactory.kt
package br.com.jiratorio.factory.domain.request

import br.com.jiratorio.domain.request.LeadTimeConfigRequest
import br.com.jiratorio.factory.KBacon
import com.github.javafaker.Faker
import org.springframework.stereotype.Component

@Component
class LeadTimeConfigRequestFactory(
    private val faker: Faker
) : KBacon<LeadTimeConfigRequest>() {

    override fun builder(): LeadTimeConfigRequest {
        return LeadTimeConfigRequest(
            name = faker.lorem().word(),
            startColumn = faker.lorem().word(),
            endColumn = faker.lorem().word()
        )
    }

}
